libraries/ms-common/src/stm32f0xx/i2c.c

```diff
@@ -5,31 +5,34 @@
 #include "stm32f0xx.h"
 
 // Arbitrary timeout
-#define I2C_TIMEOUT 1000000
-#define I2C_TIMEOUT_WHILE_FLAG(i2c, flag, status)                           \
+#define I2C_TIMEOUT 100000
+#define I2C_TIMEOUT_WHILE_FLAG(i2c_port, flag, status)                      \
   do {                                                                      \
     uint32_t timeout = (I2C_TIMEOUT);                                       \
-    while (I2C_GetFlagStatus(i2c, flag) == status) {                        \
+    while (I2C_GetFlagStatus(s_port[i2c_port].base, flag) == status) {      \
       timeout--;                                                            \
       if (timeout == 0) {                                                   \
         LOG_DEBUG("Timeout: %lu waiting for %d to change\n", flag, status); \
+        prv_recover_lockup(i2c_port);                                       \
         return status_code(STATUS_CODE_TIMEOUT);                            \
       }                                                                     \
     }                                                                       \
   } while (0)
 
-#define I2C_STOP(i2c)                                   \
-  do {                                                  \
-    I2C_TIMEOUT_WHILE_FLAG(i2c, I2C_FLAG_STOPF, RESET); \
-    I2C_ClearFlag(i2c, I2C_FLAG_STOPF);                 \
+#define I2C_STOP(i2c_port)                                   \
+  do {                                                       \
+    I2C_TIMEOUT_WHILE_FLAG(i2c_port, I2C_FLAG_STOPF, RESET); \
+    I2C_ClearFlag(s_port[i2c_port].base, I2C_FLAG_STOPF);    \
   } while (0)
 
 typedef struct {
   uint32_t periph;
   I2C_TypeDef *base;
+
```

Comment:
Nit: probably shouldn't have newlines in struct definitions. Also the struct should be defined with `struct I2CPortData`

Reply:
Hmm, I find newlines in struct definitions helps break up large structs such as in `ltc_afe.h`. I'll remove it here though.